src/seccomp_notify.c

```diff
@@ -0,0 +1,225 @@
+#define _GNU_SOURCE
+#if __STDC_VERSION__ >= 199901L
+/* C99 or later */
+#else
+#error conmon.c requires C99 or later
+#endif
+
+#include <errno.h>
+
+#include <seccomp.h>
+#include <sys/ioctl.h>
+#include <linux/seccomp.h>
+#include <sys/sysmacros.h>
+#include <dlfcn.h>
+
+#include "utils.h"
+#include "seccomp_notify.h"
+
+#ifndef SECCOMP_USER_NOTIF_FLAG_CONTINUE
+#define SECCOMP_USER_NOTIF_FLAG_CONTINUE (1UL << 0)
+#endif
+
+struct plugin {
+	void *handle;
+	void *opaque;
+	run_oci_seccomp_notify_handle_request_cb handle_request_cb;
+};
+
+struct seccomp_notify_context_s {
+	struct plugin *plugins;
+	size_t n_plugins;
+
+	struct seccomp_notif_resp *sresp;
+	struct seccomp_notif *sreq;
+	struct seccomp_notif_sizes sizes;
+};
+
+void cleanup_seccomp_notify_pluginsp(void *p)
+{
+	struct seccomp_notify_context_s **pp = p;
+	if (*pp) {
+		seccomp_notify_plugins_free(*pp);
+		*pp = NULL;
+	}
+}
+
+static inline void *xmalloc0(size_t size)
```

Comment:
can we forward declare this function and define it after it's called?

Reply:
done